runtime/Java/src/org/antlr/v4/runtime/ParserRuleContext.java

```diff
@@ -101,9 +102,8 @@ public void copyFrom(ParserRuleContext ctx) {
 			this.children = new ArrayList<>();
 			// reset parent pointer for any error nodes
 			for (ParseTree child : ctx.children) {
-				if ( child instanceof ErrorNodeImpl ) {
-					this.children.add(child);
-					((ErrorNodeImpl) child).parent = this;
+				if ( child instanceof ErrorNode ) {
+					addChild((ErrorNode)child);
```

Comment:
This is a bit weird code. You are not copying the error node, but actually create references from 2 contexts and only set the new context as parent. However, it's still in the source context, since you don't remove it from there. What is the real intention here? Really move the node or create a duplicate (like the method name suggests).